ledger-service/db-backend/src/main/scala/com/digitalasset/http/dbbackend/Queries.scala

```diff
@@ -660,16 +693,19 @@ private object OracleQueries extends Queries {
             fr" OR ",
           )
       }
-      val quotedParties = parties.toVector.map(p => s""""$p"""").mkString(", ")
-      val partiesQuery = oracleShortPathEscape(
-        '$' -: ("[*]?(@ in (": Cord) :+ quotedParties :+ "))"
-      )
-      val q =
-        sql"""SELECT c.contract_id contract_id, $tpid template_id, key, payload, signatories, observers, agreement_text
+      import Queries.CompatImplicits.catsReducibleFromFoldable1
+      val outerSelectList =
+        sql"""contract_id, template_id, key, payload,
+              signatories, observers, agreement_text"""
+      val dupQ =
+        sql"""SELECT c.contract_id contract_id, $tpid template_id, key, payload,
+                     signatories, observers, agreement_text,
+                     row_number() over (PARTITION BY c.contract_id ORDER BY c.contract_id) AS rownumber
```

Comment:
Yes, we are inventing a limit-1-per-contract-ID here. (This is the recommended approach to solving this problem without `DISTINCT`, which won't work because of the clobs, by various sources.) We used to use this prior to #9484, as well.

Reply:
ah makes sense, I misread this query on my first try